// @lekoarts/gatsby-theme-emilia/src/components/footer.tsx
/** @jsx jsx */
import { Box, jsx, Container, Flex, Link, useColorMode } from "theme-ui"
import useEmiliaConfig from "../hooks/use-emilia-config"
import SocialMediaList from "./social-media-list"
import ColorModeToggle from "./colormode-toggle"
// @ts-ignore
import AboutMeMDX from "../texts/about-me"

const Footer = () => {
  const { showThemeAuthor } = useEmiliaConfig()
  const [colorMode, setColorMode] = useColorMode()
  const isDark = colorMode === `dark`
  const toggleColorMode = (e: any) => {
    e.preventDefault()
    setColorMode(isDark ? `light` : `dark`)
  }

  return (
    <Box
      as="footer"
      variant="layout.footer"
      sx={{
        background: `linear-gradient(180deg, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, ${isDark ? `0.35` : `0.15`}) 100%)`,
      }}
    >
      <Container>
        <AboutMeMDX />
      </Container>
      <Container>
        <Flex
          sx={{
            textAlign: [`center`, `center`, `center`, `center`],
            flexDirection: `column`,
            justifyContent: `center`,
          }}
        >
          <ColorModeToggle isDark={isDark} toggle={toggleColorMode} />
          <div sx={{ mt: [4, 4, 4, 4] }}>
            <div sx={{ a: { ml: [1, 1, 1, 2], mr: [1, 1, 1, 0] } }}>
              <SocialMediaList />
            </div>
            <div sx={{ color: `textMuted` }}>Copyright &copy; {new Date().getFullYear()} Liya Khadorik. All rights reserved.</div>
          </div>
        </Flex>
      </Container>
      {showThemeAuthor && (
        <Container
          sx={{
            display: `flex`,
            justifyContent: `center`,
            alignItems: `center`,
            color: `text`,
            a: { color: `text` },
            mt: 4,
          }}
        >
          Разработка сайта:
          &nbsp;
          <Link aria-label="Link to the theme author's website" href="https://www.kopakov.com">
            Fedir Kopakov
          </Link>
        </Container>
      )}
    </Box>
  )
}

export default Footer
