// @lekoarts/gatsby-theme-emilia/src/components/seo.tsx
import React from "react"
import { Helmet } from "react-helmet"
import { withPrefix } from "gatsby"
import useSiteMetadata from "../hooks/use-site-metadata"

type SEOProps = {
  title?: string
  description?: string
  pathname?: string
  image?: string
  children?: React.ReactNode
}

const SEO = ({ title = ``, description = ``, pathname = ``, children = null }: SEOProps) => {
  const site = useSiteMetadata()

  let {
    siteTitle,
    siteTitleAlt,
    siteUrl,
    siteDescription,
    siteLanguage,
    siteImage,
    author,
    keywords,
    image,
  } = site

  const seo = {
    title,
    description,
    url: `${siteUrl}${pathname || ``}`,
    image: `${siteUrl}${image}`,
    keywords,
  }

  
  siteTitle = `Брови, ресницы Одесса - Лия Хадорик`;
  siteTitleAlt = `Брови, ресницы Одесса - Лия Хадорик`;
  siteUrl = 'https://liya.kopakov.com/';
  siteLanguage = 'ru';
  title = 'Брови, ресницы Одесса - Лия Хадорик';
  description = 'Моделирование бровей, коррекция бровей, ламинирование бровей Одесса. Брови, ресницы, ищу модель, модель Одесса. Browmaster Odessa, lashmaker odessa';
  siteDescription = 'Моделирование бровей, коррекция бровей, ламинирование бровей Одесса. Брови, ресницы, ищу модель, модель Одесса. Browmaster Odessa, lashmaker odessa';
  author = 'Лия Хадорик';
  keywords = 'Моделирование бровей, коррекция бровей, ламинирование бровей Одесса. Брови, ресницы, ищу модель, модель Одесса, Browmaster Odessa, lashmaker odessa, Брови, ресницы Одесса - Лия Хадорик, качественно, недорого, топ мастер, brow artist';
  image = 'https://liya.kopakov.com/static/banner.jpg';
  siteImage = 'https://liya.kopakov.com/static/banner.jpg';

  return (
    <Helmet title={title} titleTemplate={`%s | ${siteTitle}`}>
      <html lang={siteLanguage} />
      <meta name="description" content={description} />
      <meta name="image" content={image} />
      <meta property="og:title" content={title} />
      <meta property="og:type" content='website' />
      <meta property="og:url" content={siteUrl} />
      <meta property="og:description" content={description} />
      <meta property="keywords" content={keywords} />
      <meta property="og:image" content={image} />
      <meta property="og:type" content="website" />
      <meta property="og:image:alt" content={description} />
      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:title" content={title} />
      <meta name="twitter:url" content={siteUrl} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image" content={image} />
      <meta name="twitter:image:alt" content={description} />
      <meta name="twitter:creator" content={author} />
      <link rel="icon" type="image/png" sizes="32x32" href={withPrefix(`/favicon-32x32.png`)} />
      <link rel="icon" type="image/png" sizes="16x16" href={withPrefix(`/favicon-16x16.png`)} />
      <link rel="apple-touch-icon" sizes="180x180" href={withPrefix(`/apple-touch-icon.png`)} />
      {children}
    </Helmet>
  )
}

export default SEO
